packages/medusa/src/services/totals.ts

```diff
@@ -2,31 +2,64 @@ import _ from "lodash"
 import { BaseService } from "medusa-interfaces"
 import { MedusaError } from "medusa-core-utils"
 
+import { LineItemTaxLine } from "../models/line-item-tax-line"
+import { ShippingMethodTaxLine } from "../models/shipping-method-tax-line"
+import { Order } from "../models/order"
+import { Cart } from "../models/cart"
+import { LineItem } from "../models/line-item"
+import { Discount } from "../models/discount"
+import { DiscountRuleType } from "../models/discount-rule"
+
+import TaxProviderService from "./tax-provider"
+import { ITaxCalculationStrategy } from "../interfaces/tax-calculation-strategy"
+import { TaxCalculationContext } from "../interfaces/tax-service"
+
+import {
+  SubtotalOptions,
+  LineDiscount,
+  LineAllocationsMap,
+  LineDiscountAmount,
+} from "../types/totals"
+
+type TotalsServiceProps = {
+  taxProviderService: TaxProviderService
+  taxCalculationStrategy: ITaxCalculationStrategy
+}
+
 /**
  * A service that calculates total and subtotals for orders, carts etc..
  * @implements {BaseService}
  */
 class TotalsService extends BaseService {
```

Comment:
very nit: Most JSDoc types are off in the TotalsService, but since we now have TS, we should just remove them all.